hr_timesheet_holiday/models/hr_holidays.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+from openerp.exceptions import Warning as UserError
+
+
+class HrHolidaysStatus(models.Model):
+    """Add analytic account to holiday status"""
+    _inherit = 'hr.holidays.status'
+
+    analytic_account_id = fields.Many2one('account.analytic.account',
+        'Analytic Account')
+
+
+class HrHolidays(models.Model):
+    """Update analytic account on status change of Leave Request"""
+    _inherit = 'hr.holidays'
+
+    @api.multi
+    def holidays_validate(self):
+        res = super(HrHolidays, self).holidays_validate()
+
+        # Look for analytic account linked to Leave Type
+        for leave in self:
+            account = leave.holiday_status_id.analytic_account_id
+            if not account:
```

Comment:
I was thinking that when this module is installed, and a person forgets to put a value in "Account", and the leave is granted, there is no way to link it to the correct analytic account again. But actually, there is, you can just configure the account correctly, then refuse the leave and then grant it again. So will remove this.